pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -372,7 +372,7 @@ func popMember(members []v2pools.Member, addr string, port int) []v2pools.Member
 }
 
 func getSecurityGroupName(clusterName string, service *v1.Service) string {
-	return fmt.Sprintf("lb-sg-%s-%v", clusterName, service.Name)
+	return fmt.Sprintf("lb-sg-%s-%s-%s", clusterName, service.Namespace, service.Name)
```

Comment:
Another alternative would be to use the service UID, which is also unique over time (but less readable).

If I delete a service and then quickly create a new one with the same namespace/name - what do we want to have happen to the securityGroup?

I think we _want_ the securityGroup to not overlap in this case (so the old one can be cleaned up in parallel with the new one being created, without conflicts).
... So I think this means that this function should return
```go
fmt.Sprintf("lb-sg-%s", service.UID)
```

Reply:
>If I delete a service and then quickly create a new one with the same namespace/name - what do we want to have happen to the securityGroup?

That make sense. The service.UID is less readable.
How about "fmt.Sprintf("lb-sg-%s-%s-%s", clusterName, service.Namespace, service.Name, service.UID)"?

Reply:
That's now very long - is there any length limit that we need to worry about here?  A quick look at the code seems to imply that the name is limited to 255 chars, so I think we're ok on length.

Another (better?) option would be to use the securityGroup "description" field rather than trying to mash all our user-friendly text in the name field.  That way we can have spaces, etc and change the specific text over time without worrying about wider impact.

Personally, I think the user is going to have a pretty good idea of which cluster a securityGroup is related to, and will be able to quickly find the relevant Service based on context, ports referred to, etc without needing any additional help.

Reply:
That's right, I will check the size of its name.